ngx_http_websocket_stat_module.c

```diff
@@ -603,6 +563,88 @@ ngx_http_ws_log_format(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
     }
 }
 
+static void init_ws_log_file(ngx_conf_t *cf, ngx_http_websocket_srv_conf_t *srvcf, ngx_str_t *file_path) {
+    srvcf->ws_log = ngx_pcalloc(cf->pool, sizeof(ngx_log_t));
+
+    if (!srvcf->ws_log)
+        return;
+
+    srvcf->ws_log->log_level = NGX_LOG_NOTICE;
+    srvcf->ws_log->file = ngx_conf_open_file(cf->cycle, file_path);
+}
+
+static char *
+ngx_http_websocket_log_file(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
+{
+    ngx_http_websocket_srv_conf_t *srvcf = conf;    
+    ngx_str_t *args = cf->args->elts;
+
+    if (cf->args->nelts != 2) {
+        ngx_conf_log_error(NGX_LOG_EMERG, cf, 0, "Wrong argument number");
+        return NGX_CONF_ERROR;
+    }
+    
+    init_ws_log_file(cf, srvcf, &args[1]);
+
+    if (!srvcf->ws_log)
+        return NGX_CONF_ERROR;
+
+    if (!srvcf->ws_log->file)
+        return NGX_CONF_ERROR;
+
+    return NGX_CONF_OK;
+}
+
+static void *
+ngx_http_websocket_stat_create_srv_conf(ngx_conf_t *cf)
+{
+    ngx_http_websocket_srv_conf_t *srvcf;
+
+    srvcf = ngx_pcalloc(cf->pool, sizeof(ngx_http_websocket_srv_conf_t));
+    if (srvcf == NULL) {
```

Comment:
Earlier pointers were checked for `null` by simple negation. Don't you want to be consistent?